### PR TITLE
Vector tests

### PR DESCRIPTION
**Pull request by wbhart**
This involves a rewrite of the svector/FreeModule code to deal with some of the open issues at least for these modules.

### CODE REVIEW

File: src/module/vector.jl [hunk 1]
```diff
@@ -94,13 +98,13 @@ end
...
+   base_ring(a) != parent(b) && error("Incompatible base rings")
    R = base_ring(a)
    a1 = libSingular.p_Copy(a.ptr, R.ptr)
    b1 = libSingular.p_Copy(b.ptr, R.ptr)
    s = libSingular.p_Mult_q(a1, b1, R.ptr)
-   return Singular.Vector(R, a.rank, s)
+   return svector{spoly{T}}(R, a.rank, s)
 end
 
 *{T <: Nemo.RingElem}(a::spoly{T}, b::svector{spoly{T}}) = b*a
```
Comment (hannes14) [inline src/module/vector.jl]: Here you assume, that the module is two-sided.
How do we handle the non-commutative case (currently not in Singular.jl,
but Singular provides that) ?

**Comment by wbhart** [comment]
Thanks. What is the name of the function to multiply in the other order? Is the first function using the right Singular function?

**Comment by hannes14** [comment]
Yes, p_Mult_q(p_Copy(a,R),p_Copy(b,R),R) is a*b (a::svector,b::spoly).
The same routine works also for multiplication with ring elements from the left:
p_Mult_q(p_Copy(b,R),p_Copy(a,R),R) is b*a (b::spoly, a::svector).

**Comment by wbhart** [comment]
Should be fixed now.

**Pull request merged**